bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeGenericBindingProvider.java

```
@@ -0,0 +1,313 @@
+		boolean outBound = false;
+
+		if (bindingConfig.startsWith(IN_BOUND)) {
+			inBound = true;
+		} else if (bindingConfig.startsWith(OUT_BOUND)) {
+			outBound = true;
+		} else if (bindingConfig.startsWith(BIDIRECTIONAL)) {
+			inBound = true;
+			outBound = true;
+		} else {
+			throw new BindingConfigParseException("Item \"" + item.getName() + "\" does not start with " + IN_BOUND
+					+ ", " + OUT_BOUND + " or " + BIDIRECTIONAL + ".");
+		}
+
+		Matcher matcher = CONFIG_PATTERN.matcher(bindingConfig);
+
+		if (!matcher.matches() || matcher.groupCount() != 2)
+			throw new BindingConfigParseException("Config for item '" + item.getName() + "' could not be parsed.");
+
+		String userid = null;
+		String thermostatIdentifier = matcher.group(1);
+		if (thermostatIdentifier.contains(".")) {
+			String[] parts = thermostatIdentifier.split("\\.");
+			userid = parts[0];
+			thermostatIdentifier = parts[1];
+		}
+
+		if (inBound && !Selection.isThermostatIdentifier(thermostatIdentifier)) {
+			throw new BindingConfigParseException(
+					"Only a single thermostat identifier is permitted in an in binding or bidirectional binding.");
+		}
+
+		String property = matcher.group(2);
+
+		EcobeeBindingConfig config = new EcobeeBindingConfig(userid, thermostatIdentifier, property, inBound, outBound);
+
+		addBindingConfig(item, config);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+
+		logger.debug("validateItemType called with bindingConfig={}", bindingConfig);
+
+		Matcher matcher = CONFIG_PATTERN.matcher(bindingConfig);
+
+		if (!matcher.matches() || matcher.groupCount() != 2)
+			throw new BindingConfigParseException("Config for item '" + item.getName() + "' could not be parsed.");
+
+		String property = matcher.group(2);
+
+		logger.debug("validateItemType called with property={}", property);
+
+		/*
+		 * FIXME: find alternative to check types
```

Why alternative? This code is meant to provide early "fail fast" information if the configuration is contains wrong ItemTypes.